zoo/auditing/tasks.py

```diff
@@ -1,11 +1,21 @@
 from celery import shared_task
 
-from .models import Issue
+from .models import Issue, IssueCountByKindSnapshot, IssueCountByRepositorySnapshot
 from .utils import create_git_issue
+from django.db.models import Count
 
 
 @shared_task
 def bulk_create_git_issues(issues):
     for issue_id, user_name, reverse_url in issues:
         issue = Issue.objects.get(id=issue_id)
         create_git_issue(issue, user_name, reverse_url)
+
+
+@shared_task
+def take_issue_table_snapshots():
+    issues = Issue.objects.all()
+    for row in issues.values("repository_id", "status").annotate(count=Count("*")):
+        IssueCountByRepositorySnapshot(**row).save()
```

Comment:
I'd call `full_clean` before saving. Unless you are using a form for this, which you aren't, doing this is not ensuring that the status is one of the choices.

Reply:
eek